From d31fb39dc0c249af80b29442513f52fa788f951c Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Thu, 11 Apr 2019 12:38:28 +0300
Subject: [PATCH 543/741] dpaa2-eth: Fix incorrect version check

commit ad0e508eb862db0997701495b87e2ea329acc81d from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

The code checking whether the new autoneg API is supported
had a bug, fix it.

Signed-off-by: Catalin Neacsu <valentin-catalin.neacsu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/freescale/dpaa2/dpaa2-ethtool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/dpaa2/dpaa2-ethtool.c b/drivers/net/ethernet/freescale/dpaa2/dpaa2-ethtool.c
index 039a9879dd5f..12bf3f7ce6cf 100644
--- a/drivers/net/ethernet/freescale/dpaa2/dpaa2-ethtool.c
+++ b/drivers/net/ethernet/freescale/dpaa2/dpaa2-ethtool.c
@@ -197,7 +197,7 @@ dpaa2_eth_set_link_ksettings(struct net_device *net_dev,
 		cfg.options &= ~DPNI_LINK_OPT_HALF_DUPLEX;
 
 	if (dpaa2_eth_cmp_dpni_ver(priv, DPNI_LINK_AUTONEG_VER_MAJOR,
-				   DPNI_LINK_AUTONEG_VER_MINOR)) {
+				   DPNI_LINK_AUTONEG_VER_MINOR) < 0) {
 		err = dpni_set_link_cfg(priv->mc_io, 0, priv->mc_token, &cfg);
 	} else {
 		link_mode_ethtool2dpni(link_settings->link_modes.advertising,
-- 
2.17.1

